spec/fixtures/generate_external_certs.sh

```diff
@@ -0,0 +1,337 @@
+  if [ "$7" != "0" ]; then append_ip_altnames $7 $DIR_CONF/site_$1.conf; fi
+  
+  gen_start_end $4
+  
+  #openssl x509 -req \
+  #  -in $CSR_DIR/site_$1.csr \
+  #  -$2 \
+  #  -extensions req_ext \
+  #  -out certs/site_$1.crt \
+  #  -signkey keys/site_rsa_$1.key \
+  #  -extfile conf/site_$1.conf
+  
+  openssl ca -batch \
+    -config $DIR_CONF/site_$1.conf \
+    -out $DIR_CERTS/site_$1.crt \
+    -extensions req_ext \
+    -selfsign \
+    -keyfile $DIR_KEYS/site_rsa_$1.key \
+    -md $2 \
+    -startdate $START_DATE -enddate $END_DATE \
+    -infiles $CSR_DIR/site_$1.csr
+}
+
+# Note that an altname is specified below. If req_ext is used, then this file cannot be used as is, because an empty
+# alt_names section is not allowed. It is done this way so the file can be copied and a list of altnames
+# be appended to the end of the copy.
+function generate_conf() { # $1 ca_name
+  cat >$DIR_CONF/ca_$1.conf << EOL
+[ca]
+default_ca = default
+[default]
+certs = $DIR_CADATA/$1/
+new_certs_dir = $DIR_CADATA/$1/db.certs
+database = $DIR_CADATA/$1/db.index
+serial = $DIR_CADATA/$1/db.serial
+certificate = $DIR_CERTS/ca_$1.crt
+private_key = $DIR_KEYS/ca_$1.key
+default_days = 365
+default_crl_days = 30
+default_md = sha256
+preserve = no
+RANDFILE = $DIR_CADATA/$1/db.random
+policy = default_policy
+[default_policy]
+countryName = optional
+stateOrProvinceName = optional
+localityName = optional
+organizationName = supplied
+organizationalUnitName = supplied
+commonName = supplied
+emailAddress = optional
+[ v3_ca ]
+subjectKeyIdentifier=hash
+authorityKeyIdentifier=keyid:always,issuer
+basicConstraints = critical,CA:true
+[req_ext]
+subjectAltName = @alt_names
+[alt_names]
+EOL
+}
+
+read -r -d '' ALTNAMES_VELUM_LN << EOM
+admin.devenv.caasp.suse.net
+admin
+admin.infra.caasp.local
+aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.infra.caasp.local
+testdomain.com
+EOM
+IP_ALTNAMES_VELUM=10.17.1.0
+
+read -r -d '' ALTNAMES_KUBEAPI_LN << EOM
+kubernetes
+kubernetes.default
+kubernetes.default.svc
+kubernetes.default.svc.cluster.local
+api
+api.infra.caasp.local
+kube-api-x1.devenv.caasp.suse.net
+master-0
+master-0.infra.caasp.local
+bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
+bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb.infra.caasp.local
+EOM
+IP_ALTNAMES_KUBEAPI=172.24.0.1
+
+read -r -d '' ALTNAMES_DEX_LN << EOM
+dex
+dex.kube-system
+dex.kube-system.svc
+dex.kube-system.svc.infra.caasp.local
+dex.kube-system.svc.cluster.local
+kubernetes
+kubernetes.default
+kubernetes.default.svc
+api
+api.infra.caasp.local
+kube-api-x1.devenv.caasp.suse.net
+master-0
+master-0.infra.caasp.local
+bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
+bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb.infra.caasp.local
+EOM
+IP_ALTNAMES_DEX=172.24.0.1
+
+ALTNAMES_VELUM=$(echo "$ALTNAMES_VELUM_LN" | tr "\n" -)
+ALTNAMES_KUBEAPI=$(echo "$ALTNAMES_KUBEAPI_LN" | tr "\n" -)
+ALTNAMES_DEX=$(echo "$ALTNAMES_DEX_LN" | tr "\n" -)
+
+# Strip extra dashes at end of line
```

Comment:
Surely there is a way to avoid the extra dash??

Reply:
Maybe. But not important. Leaving it alone. It works.

Reply:
You could read into an array and then join the array using a hyphen.  When you use the `*` expansion in double quotes, the elements of the array are joined using the first char of `$IFS`.  For example:
```
sauer@lightning:~$ a=(1 2 a b pie)
sauer@lightning:~$ g=$(IFS=-; echo "${a[*]}")
sauer@lightning:~$ echo $g
1-2-a-b-pie
```

Reply:
Thanks for the explanation. Have additionally done away with all of this thanks to your help. :)